java-jsp/src/main/java/org/sonar/java/jsp/JasperOptions.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.jsp;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.Map;
+import javax.servlet.jsp.tagext.TagLibraryInfo;
+import org.apache.jasper.Options;
+import org.apache.jasper.TrimSpacesOption;
+import org.apache.jasper.compiler.JspConfig;
+import org.apache.jasper.compiler.TagPluginManager;
+import org.apache.jasper.compiler.TldCache;
+import org.apache.jasper.servlet.JspCServletContext;
+import org.apache.jasper.servlet.TldScanner;
+import org.sonar.java.AnalysisException;
+import org.xml.sax.SAXException;
+
+/**
+ * This class is replacing {@link org.apache.jasper.JspC} to avoid dependency on Ant. Most of the initialization is
+ * copied from JspC class which can be used as reference. Many of the methods are not actually used and are needed only
+ * to implement the interface.
```

Comment:
I am wondering if in order to make that statement clearer in the code, the unused method should be implemented with a `throw new UnsupportedOperationException`

Reply:
it's quite complicated to figure out what is really used, keeping the default seems safer